heartbeat/monitors/active/dialchain/dialers.go

```diff
@@ -100,7 +101,7 @@ func CreateNetDialer(timeout time.Duration) NetDialer {
 			dialer := &net.Dialer{Timeout: timeout}
 
 			start := time.Now()
-			conn, err := transport.DialWith(dialer, network, host, addresses, port)
+			conn, err := transport.DialWith(context.TODO(), dialer, network, host, addresses, port)
```

Comment:
This assumes that the incoming context will not introduce any behavior changes. Perhaps there is a deadline attached to it that is not tuned to the length of the dial calls.

I picked this option because it results in zero functional changes from what the code did before.

Reply:
Look at the calls sites you are right, nothing can call DialContext because that too was introduced to the interface. with this change